pkg/awsSGSetup/aws_security_group_setup.go

```
@@ -0,0 +1,320 @@
+	port     *int64
+	protocol string
+	srcSGId  string
+}
+
+func (is *ingressSrc) String() string {
+	if is.port == nil {
+		return fmt.Sprintf("SourceSGId: %s, Protocol: %s, Port: nil", is.srcSGId, is.protocol)
+	}
+	return fmt.Sprintf("SourceSGId: %s, Protocol: %s, Port: %d", is.srcSGId, is.protocol, *is.port)
+}
+
+// allowIngressToSG adds rules to the toSG Security Group for each element of sources.
+// Before attempting to add a rule the function checks the toSG to see if the rule already exists.
+// If there is an error adding the rules then an error is returned.
+func allowIngressToSG(cli *ec2.EC2, toSG *ec2.SecurityGroup, sources []ingressSrc) error {
+	in := &ec2.AuthorizeSecurityGroupIngressInput{}
+	sgId := aws.StringValue(toSG.GroupId)
+	in.SetGroupId(sgId)
+	for _, s := range sources {
+		log.V(DEBUG).Info("Ingress src being added", "toSG.GroupId", sgId, "ingressSrc", s.String())
+		skip := false
+		for _, x := range toSG.IpPermissions {
+			if aws.StringValue(x.IpProtocol) != s.protocol {
+				continue
+			}
+			p := aws.Int64Value(s.port)
+			if s.port != nil && (aws.Int64Value(x.FromPort) != p || aws.Int64Value(x.ToPort) != p) {
+				continue
+			}
+			for _, y := range x.UserIdGroupPairs {
+				if *y.GroupId == s.srcSGId {
+					// The Security Group already allows the traffic so skip adding this
+					// ingress rule.
+					log.V(DEBUG).Info("Ingress rule already exists", "toSG.GroupId", sgId, "ingressSrc", s.String())
+					skip = true
+					break
+				}
+			}
+			if skip {
+				break
+			}
+		}
+		if skip {
+			continue
+		}
+		in.SetIpPermissions([]*ec2.IpPermission{&ec2.IpPermission{
+			UserIdGroupPairs: []*ec2.UserIdGroupPair{{
+				GroupId: aws.String(s.srcSGId),
+			}},
+			IpProtocol: aws.String(s.protocol),
+			FromPort:   s.port,
+			ToPort:     s.port,
+		}})
+		_, err := cli.AuthorizeSecurityGroupIngress(in)
+		if err != nil {
+			log.Error(err, "Failed to add Ingress rule", "toSG", toSG, "ingressSrc", s.String())
+			return err
```

If this function returns an error, that error is ultimately printed to stdout. I think you can just return this error instead of logging it. something like...
```suggestion
			return fmt.Errorf("Failed to add Ingress rule to SG '%s' for traffic from '%s': %v", "err, toSG, s)
```